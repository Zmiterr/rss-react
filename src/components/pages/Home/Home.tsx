import React, { ReactElement } from 'react';
import Loader from 'react-loader-spinner';
import ArticleList from '../../articleList';
import Form from '../../form';
import storeSelector from '../../../hooks/useTypedSelector';
import '../../app/App.scss';

function Home(): ReactElement {
	const { pageCounter, pageSize, totalResults } = storeSelector(
		(state) => state.page
	);
	const { isLoading, isDataStatus } = storeSelector((state) => state.articles);

	return (
		<div className="App">
			<Form />

			{isDataStatus.trim() === '' ? (
				<div className="result">
					<p>
						Current page: {pageCounter}/{Math.ceil(totalResults / pageSize)}
					</p>
					<p>Total Results: {totalResults}</p>
					<p>{isDataStatus}</p>
				</div>
			) : (
				<p>{isDataStatus}</p>
			)}

			{isLoading ? (
				<div className="loader">
					<Loader type="Audio" color="#FFCBCB" height={70} width={70} />
				</div>
			) : (
				<ArticleList />
			)}
		</div>
	);
}

export default Home;
