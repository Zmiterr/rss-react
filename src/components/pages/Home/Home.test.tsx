import React from "react";
import Home from './Home';
//test deps
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter } from 'react-router-dom';



Enzyme.configure({ adapter: new Adapter() })


describe("Home component", () => {
	it("should render Home component", () => {
		const component = shallow(<BrowserRouter><Home /></BrowserRouter>);
		expect(component).toMatchSnapshot();
	});
});
