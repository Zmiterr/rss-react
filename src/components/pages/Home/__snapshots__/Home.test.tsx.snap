// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render Home component 1`] = `ShallowWrapper {}`;
